src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -7130,24 +7144,188 @@ public override IOperation VisitArgument(IArgumentOperation operation, int? capt
+
+                        // `oldInstance`
+                        var oldInstance = new FlowCaptureReferenceOperation(oldValueCaptureId, operation.Operand.Syntax,
+                            operation.Operand.Type, constantValue: operation.Operand.GetConstantValue());
+
+                        // `oldInstance.Property`
+                        var visitedValue = new PropertyReferenceOperation(property, ImmutableArray<IArgumentOperation>.Empty, oldInstance,
+                            semanticModel: null, operation.Syntax, property.Type, isImplicit: true);
+
+                        int extraValueCaptureId = GetNextCaptureId(outerCaptureRegion);
+                        AddStatement(new FlowCaptureOperation(extraValueCaptureId, operation.Syntax, visitedValue));
+
+                        LeaveRegionsUpTo(innerCaptureRegion);
+                        var extraValueCaptureRef = new FlowCaptureReferenceOperation(extraValueCaptureId, operation.Operand.Syntax,
+                            operation.Operand.Type, constantValue: operation.Operand.GetConstantValue());
+
+                        assignment = makeAssignment(property, extraValueCaptureRef, operation);
+                        initializerBuilder.Add(assignment);
+                    }
+                }
+                LeaveRegionsUpTo(outerCaptureRegion);
+
+                explicitProperties.Free();
+
+                return new AnonymousObjectCreationOperation(initializerBuilder.ToImmutableAndFree(), semanticModel: null, operation.Syntax, operation.Type, operation.IsImplicit);
+            }
+
+            // Build an operation for `<implicitReceiver>.Property = <capturedValue>`
+            SimpleAssignmentOperation makeAssignment(IPropertySymbol property, IOperation capturedValue, WithOperation operation)
+            {
+                // <implicitReceiver>
+                var implicitReceiver = new InstanceReferenceOperation(InstanceReferenceKind.ImplicitReceiver,
                     semanticModel: null, operation.Syntax, operation.Type, isImplicit: true);
 
-            return PopStackFrame(frame, HandleObjectOrCollectionInitializer(operation.Initializer, cloned));
+                // <implicitReceiver>.Property
+                var target = new PropertyReferenceOperation(property, ImmutableArray<IArgumentOperation>.Empty, implicitReceiver,
+                    semanticModel: null, operation.Syntax, property.Type, isImplicit: true);
+
+                // <implicitReceiver>.Property = <capturedValue>
+                return new SimpleAssignmentOperation(isRef: false, target, capturedValue,
+                    semanticModel: null, operation.Syntax, property.Type, constantValue: null, isImplicit: true);
+            }
+
+            static bool setAllProperties(ImmutableArray<IOperation> initializers, ImmutableArray<IPropertySymbol> properties)
+            {
+                var set = PooledHashSet<string>.GetInstance();
```

Comment:
>string [](http://example.com/codeflow?start=40&length=6)

Why use name as the key instead of the property symbol?

Reply:
Name seems simpler.
Yes, it's possible to have duplicate names in error scenarios and this code handles that (we ignore the return value from `Add` and `Remove`)